packages/adapter-mongoose/lib/adapter-mongoose.js

```diff
@@ -84,6 +86,19 @@ class MongooseAdapter extends BaseKeystoneAdapter {
     const { MongoId } = require('@keystone-alpha/fields-mongoid');
     return MongoId.primaryKeyDefaults[this.name].getConfig();
   }
+
+  checkDatabaseVersion() {
+    new this.mongoose.mongo.Admin(this.mongoose.connection.db).buildInfo((error, info) => {
+      if (error) {
+        console.log(`Error reading version from MongoDB: ${error}`);
+      }
+      if (!versionGreaterOrEqualTo(info.version, this.minVer)) {
+        throw new Error(
+          `MongoDB version ${info.version} is incompatible. Version ${this.minVer} or later is required.`
+        );
+      }
+    });
+  }
```

Comment:
I'm not sure about the semantics of the `buildInfo` function, but I imagine it's always async. Therefore `checkDatabseVersion` should also be async:

```javascript
  checkDatabaseVersion() {
    return new Promise((resolve, reject) => {
      new this.mongoose.mongo.Admin(this.mongoose.connection.db).buildInfo((error, info) => {
        if (error) {
          console.log(`Error reading version from MongoDB: ${error}`);
        }
        if (!versionGreaterOrEqualTo(info.version, this.minVer)) {
          reject(new Error(
            `MongoDB version ${info.version} is incompatible. Version ${this.minVer} or later is required.`
          ));
        }
        resolve();
      });
    });
  }
```

This will also work nicely with the `try/catch` in `packages/keystone/lib/adapters/index.js`

Reply:
Did some digging. `buildInfo` either takes a callback (as I was using) or returns a promise if no callback was provided. Switched to the promise impl (bit different from your suggestion).